### PR TITLE
refactor: replace nil checks with not in conditions

### PR DESCRIPTION
**Pull request by omarcopires**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed 
This commit introduces a refactor to replace nil checks with the use of not in logical conditions, aligning with the preferred style in the codebase. This change improves code readability and consistency throughout the affected sections.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by Zbizu** [comment]
this change introduces `nil` comparisons which tfs moved away from, except for situations when both nil and false can be returned by a function and need to be distinguished

if the statement does not need to distinguish between false and nil:
- `~= nil` is not necessary in logical statements
- `if x == nil then` should be replaced with `if not x then`

**Comment by MillhioreBT** [comment]
What @Zbizu says is true, please make the change from `if not x then`
and for cases of setStorageValue(-1), change to removeStorageValue

File: data/creaturescripts/scripts/login.lua [hunk 1]
```diff
@@ -13,20 +13,17 @@ function onLogin(player)
 	-- Promotion
 	local vocation = player:getVocation()
 	local promotion = vocation:getPromotion()
-	if player:isPremium() then
-		local value = player:getStorageValue(PlayerStorageKeys.promotion)
-		if value == 1 then
-			player:setVocation(promotion)
-		end
-	elseif not promotion then
+	if player:isPremium() and player:getStorageValue(PlayerStorageKeys.promotion) == 1 then
```
Comment (MillhioreBT) [inline data/creaturescripts/scripts/login.lua:16]: The value here can be null, leave it as it was, just add `value and value == 1`

**Review by MillhioreBT — CHANGES_REQUESTED** [review]
Change what you check on line 16 in login.lua

**Pull request merged**